### DISCUSSION TITLE
DongShanPi One

### DISCUSSION THREAD

**Discussion by fifteenhex** [linux-chenxing/linux-chenxing.org#47] [General]
They sent me some boards to get this working and seem to want mainline support to work out.

Wiki page for this board started here:
http://linux-chenxing.org/infinity2/dongshanpione/

**Comment by codebug8** [comment]
### dongshanpi one reference compiling and flashing u-boot cannot start

I obtained and compiled the u-boot source code by referring to the tutorial, but I found that there was no response output when I started it after programming. The following figure shows the file I compiled and output. In it, I refer to the ido-som2d01 article to compile the compiled ipl The file u-boot-dtb.img is exported and burned as shown in the figure below.
http://linux-chenxing.org/infinity2/dongshanpione/
http://linux-chenxing.org/infinity2/ido-som2d01/#flashing
 
After finishing the flashing command, I cannot find the corresponding bin file from your uboot warehouse for GCIS.bin and IPL.bin files, so I directly use the two files in the SigMstar flashing tool boot, including the IPL.bin file My size here is 22KB, but I think the size of your flashing partition is 21KB, so I guess the IPL.bin we use is different, which leads to the problem of not being able to start later.
If you are free, can you provide the GCIS.bin and IPL.bin files that you can start. I will try to flash and start again.

sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -e
sudo SNANDer -p mstarddc -c /dev/i2c-0:49  -w GCIS.bin
sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -a 0x140000 -l 0x54C0 -w IPL.bin
sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -a 0x200000 -l 0xBF58  -w ipl
sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -a 0x2C0000 -l 0x64000 -w u-boot-dtb.img
 
Serial port startup output after programming is completed

**Reply by fifteenhex** [reply]
Hi,

Give me 20 minutes... I'm retesting on mine.

**Reply by codebug8** [reply]
Hi Daniel
After repeated verifications, I found that using Sigmstar Flash Tool to burn GCIS.bin IPL.bin idosom2d01-ipl.bin u-boot-dtb.img (dongshanpi configuration production) can start and output.
I still don’t understand why there is no response when I use the SNANDer flashing tool to flash under Ubuntu.

**Reply by fifteenhex** [reply]
There was a missing option for serial in the SPL. Can you force update your u-boot branch and try again.
I have added some extra info to the wikipage (http://linux-chenxing.org/infinity2/dongshanpione/)

**Reply by codebug8** [reply]
Ok i'll try again thanks

**Reply by codebug8** [reply]
> There was a missing option for serial in the SPL. Can you force update your u-boot branch and try again. I have added some extra info to the wikipage (http://linux-chenxing.org/infinity2/dongshanpione/)

I found that we didn’t make any difference when erasing, but when flashing files, I always have a hint here.
  Error sending read command: errno 5, tries left 10
This may be the problem, but it’s the first time I came into contact with this programming method, and I don’t understand why this problem occurs. The device driver I use is https://github.com/gschorcht/i2c-ch341-usb
```shell
book@100ask:~/DongshanPi$ lsusb
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 004: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
Bus 003 Device 003: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
Bus 003 Device 005: ID 1a86:5512 QinHeng Electronics CH341 in EPP/MEM/I2C mode, EPP/I2C adapter
Bus 003 Device 002: ID 0e0f:0003 VMware, Inc. Virtual Mouse
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 002: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub

```
Among them, the CH341AUSB to I2C tool is https://item.taobao.com/item.htm?id=616744148415. I tried to modify the I2C programming rate but all ended in failure. Have you encountered similar problems?
``` shell
book@100ask:~/DongshanPi$ sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -e

SNANDer - Serial Nor/nAND/Eeprom programmeR v.1.7.2 by McMCC <mcmcc@mail.ru>

connection /dev/i2c-0:49
Info: Will try to use device /dev/i2c-0 and address 0x49.
Info: Will NOT reset the device at the end.
here xx1
here xx2
spi_nand_probe: mfr_id = 0x9c, dev_id = 0x1
Get Status Register 1: 0x00
Get Status Register 2: 0x10
Using Flash ECC.
Detected SPI NAND Flash: XINCUN XCSP1AAWH-NT, Flash Size: 128 MB
ERASE:
Set full erase chip!
Erase addr = 0x0000000000000000, len = 0x0000000008000000
spi_nand_erase_block : erase block fail, block = 0x200, status = 0x4
spi_nand_erase_internal : Erase Fail at addr = 0x4000000, len = 0x8000000, block_idx = 0x200
spi_nand_erase_block : erase block fail, block = 0x201, status = 0x4
spi_nand_erase_internal : Erase Fail at addr = 0x4020000, len = 0x8000000, block_idx = 0x201
spi_nand_erase_block : erase block fail, block = 0x203, status = 0x4
spi_nand_erase_internal : Erase Fail at addr = 0x4060000, len = 0x8000000, block_idx = 0x203
spi_nand_erase_block : erase block fail, block = 0x300, status = 0x4
spi_nand_erase_internal : Erase Fail at addr = 0x6000000, len = 0x8000000, block_idx = 0x300
spi_nand_erase_block : erase block fail, block = 0x3ff, status = 0x4
spi_nand_erase_internal : Erase Fail at addr = 0x7fe0000, len = 0x8000000, block_idx = 0x3ff
Erase 100% [134217728] of [134217728] bytes      
Status: BAD(-1)
```

``` shell
book@100ask:~/DongshanPi$ sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -w GCIS.bin 

SNANDer - Serial Nor/nAND/Eeprom programmeR v.1.7.2 by McMCC <mcmcc@mail.ru>

connection /dev/i2c-0:49
Info: Will try to use device /dev/i2c-0 and address 0x49.
Info: Will NOT reset the device at the end.
here xx1
here xx2
spi_nand_probe: mfr_id = 0x9c, dev_id = 0x1
Get Status Register 1: 0x00
Get Status Register 2: 0x11
Using Flash ECC.
Detected SPI NAND Flash: XINCUN XCSP1AAWH-NT, Flash Size: 128 MB
WRITE:
Write addr = 0x0000000000000000, len = 0x0000000000000800
Error sending read command: errno 5, tries left 10
Error sending read command: errno 5, tries left 9
Error sending read command: errno 5, tries left 8
Error sending read command: errno 5, tries left 7
Error sending read command: errno 5, tries left 6
Error sending read command: errno 5, tries left 5
Error sending read command: errno 5, tries left 4
Error sending read command: errno 5, tries left 3
Error sending read command: errno 5, tries left 2
Error sending read command: errno 5, tries left 1
Error sending write command: errno 5.
Written 100% [2048] of [2048] bytes      
Elapsed time: 0 seconds
Status: OK
book@100ask:~/DongshanPi$ 

```
dmesg  commad log 
``` shell 
ch341_i2c_transfer: size of data is too large for existing USB endpoints
```
I think it is caused by a problem with the CH341A writer chip。

**Reply by fifteenhex** [reply]
mm I guess the ch341a can't handle the size of transfer being used. :(

**Reply by codebug8** [reply]
What is the Flash writer you use?

**Reply by fifteenhex** [reply]
I use a digispark flashed with this firmware: https://github.com/harbaum/I2C-Tiny-USB/tree/master/digispark

**Reply by codebug8** [reply]
Thank you very much 🍻

**Reply by ltrsunny** [reply]
> > There was a missing option for serial in the SPL. Can you force update your u-boot branch and try again. I have added some extra info to the wikipage (http://linux-chenxing.org/infinity2/dongshanpione/)
> 
> I found that we didn’t make any difference when erasing, but when flashing files, I always have a hint here. Error sending read command: errno 5, tries left 10 This may be the problem, but it’s the first time I came into contact with this programming method, and I don’t understand why this problem occurs. The device driver I use is https://github.com/gschorcht/i2c-ch341-usb
> 
> ```shell
> book@100ask:~/DongshanPi$ lsusb
> Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
> Bus 003 Device 004: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
> Bus 003 Device 003: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
> Bus 003 Device 005: ID 1a86:5512 QinHeng Electronics CH341 in EPP/MEM/I2C mode, EPP/I2C adapter
> Bus 003 Device 002: ID 0e0f:0003 VMware, Inc. Virtual Mouse
> Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
> Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
> Bus 002 Device 002: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
> Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
> ```
> 
> Among them, the CH341AUSB to I2C tool is https://item.taobao.com/item.htm?id=616744148415. I tried to modify the I2C programming rate but all ended in failure. Have you encountered similar problems?
> 
> ```shell
> book@100ask:~/DongshanPi$ sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -e
> 
> SNANDer - Serial Nor/nAND/Eeprom programmeR v.1.7.2 by McMCC <mcmcc@mail.ru>
> 
> connection /dev/i2c-0:49
> Info: Will try to use device /dev/i2c-0 and address 0x49.
> Info: Will NOT reset the device at the end.
> here xx1
> here xx2
> spi_nand_probe: mfr_id = 0x9c, dev_id = 0x1
> Get Status Register 1: 0x00
> Get Status Register 2: 0x10
> Using Flash ECC.
> Detected SPI NAND Flash: XINCUN XCSP1AAWH-NT, Flash Size: 128 MB
> ERASE:
> Set full erase chip!
> Erase addr = 0x0000000000000000, len = 0x0000000008000000
> spi_nand_erase_block : erase block fail, block = 0x200, status = 0x4
> spi_nand_erase_internal : Erase Fail at addr = 0x4000000, len = 0x8000000, block_idx = 0x200
> spi_nand_erase_block : erase block fail, block = 0x201, status = 0x4
> spi_nand_erase_internal : Erase Fail at addr = 0x4020000, len = 0x8000000, block_idx = 0x201
> spi_nand_erase_block : erase block fail, block = 0x203, status = 0x4
> spi_nand_erase_internal : Erase Fail at addr = 0x4060000, len = 0x8000000, block_idx = 0x203
> spi_nand_erase_block : erase block fail, block = 0x300, status = 0x4
> spi_nand_erase_internal : Erase Fail at addr = 0x6000000, len = 0x8000000, block_idx = 0x300
> spi_nand_erase_block : erase block fail, block = 0x3ff, status = 0x4
> spi_nand_erase_internal : Erase Fail at addr = 0x7fe0000, len = 0x8000000, block_idx = 0x3ff
> Erase 100% [134217728] of [134217728] bytes      
> Status: BAD(-1)
> ```
> 
> ```shell
> book@100ask:~/DongshanPi$ sudo SNANDer -p mstarddc -c /dev/i2c-0:49 -w GCIS.bin 
> 
> SNANDer - Serial Nor/nAND/Eeprom programmeR v.1.7.2 by McMCC <mcmcc@mail.ru>
> 
> connection /dev/i2c-0:49
> Info: Will try to use device /dev/i2c-0 and address 0x49.
> Info: Will NOT reset the device at the end.
> here xx1
> here xx2
> spi_nand_probe: mfr_id = 0x9c, dev_id = 0x1
> Get Status Register 1: 0x00
> Get Status Register 2: 0x11
> Using Flash ECC.
> Detected SPI NAND Flash: XINCUN XCSP1AAWH-NT, Flash Size: 128 MB
> WRITE:
> Write addr = 0x0000000000000000, len = 0x0000000000000800
> Error sending read command: errno 5, tries left 10
> Error sending read command: errno 5, tries left 9
> Error sending read command: errno 5, tries left 8
> Error sending read command: errno 5, tries left 7
> Error sending read command: errno 5, tries left 6
> Error sending read command: errno 5, tries left 5
> Error sending read command: errno 5, tries left 4
> Error sending read command: errno 5, tries left 3
> Error sending read command: errno 5, tries left 2
> Error sending read command: errno 5, tries left 1
> Error sending write command: errno 5.
> Written 100% [2048] of [2048] bytes      
> Elapsed time: 0 seconds
> Status: OK
> book@100ask:~/DongshanPi$ 
> ```
> 
> dmesg commad log
> 
> ```shell
> ch341_i2c_transfer: size of data is too large for existing USB endpoints
> ```
> 
> I think it is caused by a problem with the CH341A writer chip。

What is your OS to use this CH341 driver? I am trying it on Fedora 35 but got an `ERROR: modpost: "irq_to_desc" [/home/dev/i2c-ch341-usb/i2c-ch341-usb.ko] undefined!`

**Reply by codebug8** [reply]
I tested ch341a before and found that it cannot be used normally

**Comment by codebug8** [comment]
Hello, I don't quite understand why SSD202D can't directly use the programming tool to program the uboot lts ipl part and the u-boot.img part, but to use the serial port to program and load separately, can you help me understand? Because many of our students have reported that this method is not suitable for operations such as later mass production.

**Reply by fifteenhex** [reply]
It is possible now. Writing the ubi image to the right place will allow the dongshanpione to boot fully into u-boot without serial loading.

**Reply by codebug8** [reply]
Can you tell me what should I do? At the same time, I also hope that you will take a look at the mailbox. We have a new communication with Sigmastar, and we are going to try to make SSD222D SSD268G. I hope you can provide us with some suggestions and help.

**Reply by fifteenhex** [reply]
I'll give some instructions when I get home.. It's just writing the ubi to the right address though.
I saw your mail. I will reply today :)

**Reply by codebug8** [reply]
Thank you！
